lib/runBlock.js

```diff
@@ -217,13 +227,11 @@ module.exports = function (opts, cb) {
           if (ethUtil.bufferToInt(block.header.gasUsed) !== Number(gasUsed)) {
             err = new Error((err || '') + 'invalid gasUsed ')
           }
-          if (self.stateManager.trie.root.toString('hex') !== block.header.stateRoot.toString('hex')) {
+          if (stateRoot.toString('hex') !== block.header.stateRoot.toString('hex')) {
```

Comment:
Extra failures are happening here, not sure why yet, probably you'll get this faster than me.

When I run ``node ./tests/tester.js -b --file='suicideStorageCheck'``, looking at the first test and and add console output for the first and second state root value, I get:

```
211aca80b1d1084fd9a18fda2afbc5bdf3866d06c5321660e105546e79fded8d
334d8d12b70ab372703f37bf6c9fed76c7c61ba1f131517123d5d440776d825a
```

On the old version I get both:

```
334d8d12b70ab372703f37bf6c9fed76c7c61ba1f131517123d5d440776d825a
334d8d12b70ab372703f37bf6c9fed76c7c61ba1f131517123d5d440776d825a
```